### PR TITLE
Add loads for certain py/java rules

### PR DESCRIPTION
**Pull request by ted-xie**
Bazel 8 removes native shell, python, and java rules. This PR makes rules_robolectric compatible with Bazel 8.

### CODE REVIEW

**Comment by ted-xie** [comment]
Provided this PR gets merged, we (the Bazel team) would also need a release cut relatively quickly in order to unblock the next rules_android release, which will unblock the upcoming Bazel 8 release.

**Review by Bencodes — APPROVED** [review]

**Comment by ted-xie** [comment]
@Bencodes @wyverald There ended up being some kind of java [toolchain issue](https://github.com/robolectric/robolectric-bazel/actions/runs/12057013955/job/33620959524) when I tried upgrading rules_java to 8.5.1 for the example app:

```
ERROR: /home/runner/.cache/bazel/_bazel_runner/faa3a579053c5258adca3d7c148553b6/external/rules_java_builtin/toolchains/BUILD:254:14: While resolving toolchains for target @@rules_java_builtin//toolchains:platformclasspath (b8277ad): No matching toolchains found for types @@bazel_tools//tools/jdk:bootstrap_runtime_toolchain_type.
```

Note that this _only_ occured with WORKSPACE mode. I tried debugging this for a while but wasn't able to come up with anything concrete. I suspect it's something to do with the native Android rules in rules_android v0.1.1 interacting badly with the latest rules_java, but upgrading the Android rule dep in WORKSPACE was unsavory to say the least.

The only reason I wanted to upgrade rules_java in the first place was to get the rules_java version of http_jar(), but that involved pulling in a lot of complicated WORKSPACE setup and the aforementioned toolchain issue. The next best solution I could think of was to copy over the (fairly standalone) http_jar() rule into robolectric itself, which is also gross, but much less complicated than making WORKSPACE builds work.

File: bazel/java/http_jar.bzl [hunk 1]
```diff
@@ -0,0 +1,200 @@
+"""The http_jar repo rule, for downloading jars over HTTP.
+
+### Setup
```
[the remark this answers is not in the document]
Comment (ted-xie) [reply bazel/java/http_jar.bzl:3]: Yeah, I would prefer not copying this file at all. It actually has a new home in rules_java v8.5.0, but I had a lot of trouble getting that rules_java version working with robolectric-bazel in WORKSPACE mode.

File: bazel/java/http_jar.bzl [hunk 2]
```diff
@@ -0,0 +1,200 @@
+"""The http_jar repo rule, for downloading jars over HTTP.
```
[the remark this answers is not in the document]
Comment (Bencodes) [reply bazel/java/http_jar.bzl:1]: https://github.com/robolectric/robolectric-bazel/pull/122

**Pull request merged**